### PR TITLE
LogBatchPurger with DML row limits in mind

### PR DESCRIPTION
**Pull request by jamessimone**
* potential fix for #166 by introducing limits-minded deletion mechanism in LogBatchPurger

@jongpie no big attachment to this, but I saw this issue come through and thought I might submit something for it. Let me know what you think!

### CODE REVIEW

File: nebula-logger/main/log-management/classes/LogBatchPurger.cls [hunk 1]
```diff
@@ -9,12 +9,60 @@
...
+        private final List<SObject> recordsToDelete;
+        public LogDeleter(List<SObject> recordsToDelete) {
+            this.recordsToDelete = recordsToDelete;
+        }
+
+        public void process() {
+            if (DELETED_COUNT + this.recordsToDelete.size() < MAX_RECORDS_DELETED) {
+                this.hardDelete(this.recordsToDelete);
+                this.recordsToDelete.clear();
+            } else {
```
Comment (jongpie) [inline nebula-logger/main/log-management/classes/LogBatchPurger.cls:38]: It looks like [in codecov](https://app.codecov.io/gh/jongpie/NebulaLogger/compare/167/diff#diff-bmVidWxhLWxvZ2dlci9tYWluL2xvZy1tYW5hZ2VtZW50L2NsYXNzZXMvTG9nQmF0Y2hQdXJnZXIuY2xz), there's not any code coverage for this `else` statement - thoughts on marking `MAX_RECORDS_DELETED` as `@testVisible` --> adding a test that verifies that the queueable job is enqueued?

**Review by jongpie — CHANGES_REQUESTED** [review]
@jamessimone this looks great! I really like the approach - I just added one comment about some code coverage.

File: nebula-logger/main/log-management/classes/LogBatchPurger.cls (hunk 1 above)
Comment (jamessimone) [reply nebula-logger/main/log-management/classes/LogBatchPurger.cls:38]: Iiiinteresting! Funny story - I had a test for this, and the lines covered, but then I removed it. That was a total mistake on my part, I’ll rewrite it tomorrow and include it here.

Comment (jongpie) [reply nebula-logger/main/log-management/classes/LogBatchPurger.cls:38]: Sounds like a plan! Thanks!

**Pull request merged**